Request/ParamFetcher.php

```diff
@@ -11,15 +11,21 @@
 
```

Comment:
I am wondering if we do not need to refactor this a bit to bring the methods back down to a manageable level of complexity .. quite some deep if constructs we have here ..